include/rnp/rnp2.h

```diff
@@ -260,6 +264,54 @@ rnp_result_t rnp_key_have_public(rnp_key_handle_t key, bool *result);
 
 /* TODO define functions for password-based encryption */
 
+rnp_result_t rnp_op_sign_create(rnp_op_sign_t *op,
+                                rnp_ffi_t      ffi,
+                                rnp_input_t    input,
+                                rnp_output_t   output);
+
+rnp_result_t rnp_op_sign_set_signing_key(rnp_op_sign_t op, rnp_key_handle_t key);
+
+rnp_result_t rnp_op_sign_set_compression(rnp_op_sign_t op, const char *compression, int level);
+rnp_result_t rnp_op_sign_set_armor(rnp_op_sign_t op, bool armored, bool cleansign);
+rnp_result_t rnp_op_sign_set_detached(rnp_op_sign_t op, bool detached);
+
+rnp_result_t rnp_op_sign_set_timestamps(rnp_op_sign_t op,
+                                        uint32_t      creation_time,
+                                        uint32_t      expiration_time);
+rnp_result_t rnp_op_sign_set_hash_fn(rnp_op_sign_t op, const char *hash);
```

Comment:
I think naming this `rnp_op_sign_set_hash` would align better with the json key generation and key listing stuff.

Reply:
Included hash function as an arg to `rnp_op_sign_add_signer` instead since it's totally reasonable when multi singing to use different hashes for different keys (eg SHA-512 for RSA + SHA-256 for P-256 ECDSA).